### PR TITLE
[INLONG-2382][Sort-sdk] Support Pulsar consumer of PB compression cache message protocol

### LINKED ISSUE #2382: [Feature] Sort-sdk support Pulsar consumer of PB compression cache message protocol.
### Description

 Sort-sdk support Pulsar consumer of new cache message protocol.
Parent issue: #2375 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wardlican**
[INLONG-2382]  Sort-sdk support Pulsar consumer of PB compression cache message protocol

### Title Name: [INLONG-2382]  [Sort-sdk] support Pulsar consumer of PB compression cache message protocol

where *XYZ* should be replaced by the actual issue number.

Fixes #2382

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by dockerzhang** [comment]
@luchunliang PTAL

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+     * @return {@link List< InLongMessage >}
+     */
+    private List<InLongMessage> transformMessageObjs(ClientContext context, InLongTopic inLongTopic,
+            MessageObjs messageObjs, String inlongGroupId,
+            String inlongStreamId) {
+        List<InLongMessage> inLongMessages = new ArrayList<>();
+        for (MessageObj messageObj : messageObjs.getMsgsList()) {
+            List<MapFieldEntry> mapFieldEntries = messageObj.getParamsList();
+            Map<String, String> headers = new HashMap<>();
+            for (MapFieldEntry mapFieldEntry : mapFieldEntries) {
```
Comment (luchunliang) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:130]: mapFieldEntries maybe be null.

File: inlong-sdk/sort-sdk/pom.xml [hunk 2]
```diff
@@ -113,15 +132,9 @@
...
-            <version>${project.version}</version>
-        </dependency>
-
-        <dependency>
-            <groupId>org.apache.inlong</groupId>
-            <artifactId>tubemq-client</artifactId>
-            <version>${project.version}</version>
+            <groupId>org.xerial.snappy</groupId>
+            <artifactId>snappy-java</artifactId>
+            <version>1.1.7.3</version>
```
Comment (luchunliang) [inline inlong-sdk/sort-sdk/pom.xml:137]: Artifact version information need to move to "properties".

**Review by luchunliang — APPROVED** [review]

File: inlong-sdk/sort-sdk/pom.xml (hunk 2 above)
Comment (wardlican) [reply inlong-sdk/sort-sdk/pom.xml:137]: ok

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java (hunk 1 above)
Comment (wardlican) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:133]: yes mapFieldEntries maybe be null

File: inlong-sdk/sort-sdk/pom.xml [hunk 3]
```diff
@@ -76,13 +76,32 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <plugin.compile.version>3.8.1</plugin.compile.version>
+        <pulsar.version>2.7.2</pulsar.version>
```
[the remark this answers is not in the document]
Comment (wardlican) [reply inlong-sdk/sort-sdk/pom.xml]: ok i will improve it to 2.8.1

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**